services/cloud/node/node.go

```diff
@@ -74,3 +78,23 @@ func getNodeStatuses(node *wssdcloud.Node) map[string]*string {
 	statuses["Info"] = convert.ToStringPtr(node.GetInfo().String())
 	return statuses
 }
+
+func generateNodeTags(node *wssdcloud.Node) map[string]*string {
+	tags := make(map[string]*string)
+	populateOsRegistrationStatusTag(tags, node)
+	if len(tags) > 0 {
+		return tags
+	}
+	return nil
+}
+
+func populateOsRegistrationStatusTag(tags map[string]*string, node *wssdcloud.Node) {
+	if node.Info != nil {
```

Comment:
[nitpick] you could merge the three consecutive "if" statements into one, so that you lower the level of imbrications and make the code easier to read.